src/web_peer_connection.cpp

```diff
@@ -86,16 +86,20 @@ web_peer_connection::web_peer_connection(peer_connection_args const& pack
 	shared_ptr<torrent> tor = pack.tor.lock();
 	TORRENT_ASSERT(tor);
 
-	// we always prefer downloading 1 MiB chunks
-	// from web seeds, or whole pieces if pieces
-	// are larger than a MiB
-	int preferred_size = 1024 * 1024;
+	// we prefer downloading large chunks from web seeds,
+	// but still want to be able to split requests
+	int preferred_size = m_settings.get_int(settings_pack::urlseed_max_request_bytes);
```

Comment:
I think ``preferred_size`` should be const here, which means you have to initialize it with a ``std::max()`` expression and calculate ``min_size`` before getting the settings value. In fact, I think ``min_size`` should also be const and initialized only once, ``int const min_size = std::max((web.supports_keepalive ? 1 : 4) * 1024 * 1024, tor->torrent_file().piece_length());``